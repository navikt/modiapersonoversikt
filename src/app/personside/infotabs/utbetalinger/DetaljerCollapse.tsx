import * as React from 'react';
import { ReactNode } from 'react';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { UnmountClosed } from 'react-collapse';
import { FlexEnd } from '../../../../components/common-styled-components';
import NavFrontendChevron from 'nav-frontend-chevron';
import { Normaltekst } from 'nav-frontend-typografi';

interface Props {
    open: boolean;
    toggle: () => void;
    children: ReactNode;
}

const Wrapper = styled<{ open: boolean }, 'div'>('div')`
  transition: .5s;
  padding: 0 ${theme.margin.px20} ${theme.margin.px20};
  ${props => props.open && theme.ekspandert};
`;

const CollapseAnimasjon = styled<{ open: boolean }, 'div'>('div')`
  transition: .5s;
  padding: 0 ${theme.margin.px20};
  ${props => props.open && 'padding: 0'};
`;

const KnappWrapper = styled<{ open: boolean }, 'button'>('button')`
  border: none;
  padding: .1rem .2rem;
  margin-top: ${props => props.open && '1rem'};
  border-radius: ${theme.borderRadius.knapp};
  cursor: pointer;
  background-color: transparent;
  display: flex;
  align-items: center;
  color: ${theme.color.lenke};
  &:focus {
    ${theme.focus}
  }
  &:hover {
    opacity: 0.8;
    text-decoration: underline;
  }
 @media print {
    display: none;
 }
`;

const Luft = styled.span`
  margin-right: .5rem;
`;

interface KnappProps {
    onClick: (event: React.MouseEvent<HTMLButtonElement>) => void;
    open: boolean;
}

function DetaljerKnapp(props: KnappProps) {
    return (
        <KnappWrapper
            onClick={props.onClick}
            aria-expanded={props.open}
            open={props.open}
        >
            <Luft>
                <Normaltekst tag="span">
                    {props.open ? 'Skjul' : 'Vis'} detaljer
                </Normaltekst>
            </Luft>
            <NavFrontendChevron type={props.open ? 'opp' : 'ned'}/>
        </KnappWrapper>
    );
}

function DetaljerCollapse(props: Props) {
    return (
        <Wrapper open={props.open}>
            <CollapseAnimasjon open={props.open}>
                <UnmountClosed isOpened={props.open}>
                    {props.children}
                </UnmountClosed>
            </CollapseAnimasjon>
            <FlexEnd>
                <DetaljerKnapp
                    onClick={props.toggle}
                    open={props.open}
                />
            </FlexEnd>
        </Wrapper>
    );
}

export default DetaljerCollapse;
