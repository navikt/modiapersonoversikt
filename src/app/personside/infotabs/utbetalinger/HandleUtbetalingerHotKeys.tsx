import * as React from 'react';
import { Utbetaling, Ytelse } from '../../../../models/utbetalinger';
import { flatMapYtelser } from './utils/utbetalingerUtils';
import { AnyAction, Dispatch } from 'redux';
import {
    setEkspanderYtelse,
    setNyYtelseIFokus
} from '../../../../redux/utbetalinger/utbetalingerStateReducer';
import { AppState } from '../../../../redux/reducers';
import { connect } from 'react-redux';
import styled from 'styled-components';

interface OwnProps {
    children: React.ReactNode;
    utbetalinger: Utbetaling[];
}

interface DispatchProps {
    settYtelseIFokus: (ytelse: Ytelse | null) => void;
    ekspanderYtelse: (ytelse: Ytelse, ekspander: boolean) => void;
}

interface StateProps {
    ytelseIFokus: Ytelse | null;
    ekspanderteYtelser: Ytelse[];
}

type Props = DispatchProps & OwnProps & StateProps;

function eventTargetIsButton(event: React.KeyboardEvent) {
    return event.target instanceof HTMLButtonElement;
}

const Wrapper = styled.div`
  &:focus {
    outline: none;
  }
`;

class HandleUtbetalingerHotKeys extends React.Component<Props> {

    constructor(props: Props) {
        super(props);
        this.handleKeyDown = this.handleKeyDown.bind(this);
    }

    handleKeyDown(event: React.KeyboardEvent) {
        switch (event.which) {
            case 13:
                this.handleEnter(event);
                break;
            case 32:
                this.handleSpace(event);
                break;
            case 38:
                this.settForrigeYtelseIFokus();
                break;
            case 40:
                this.settNesteYtelseIFokus();
                break;
            default:
        }
    }

    settNesteYtelseIFokus() {
        const ytelser: Ytelse[] = flatMapYtelser(this.props.utbetalinger);
        const currentIndex = this.props.ytelseIFokus ? ytelser.indexOf(this.props.ytelseIFokus) : -1;
        const nesteYtelse = ytelser[currentIndex + 1] || ytelser[0];
        this.props.settYtelseIFokus(nesteYtelse);
    }

    settForrigeYtelseIFokus() {
        const ytelser: Ytelse[] = flatMapYtelser(this.props.utbetalinger);
        const currentIndex = this.props.ytelseIFokus ? ytelser.indexOf(this.props.ytelseIFokus) : ytelser.length;
        const forrigeYtelse = ytelser[currentIndex - 1] || ytelser[ytelser.length - 1];
        this.props.settYtelseIFokus(forrigeYtelse);
    }

    handleEnter(event: React.KeyboardEvent) {
        if (event.repeat || eventTargetIsButton(event)) {
            return;
        }
        this.toggleEkspanderYtelseIFokus();
    }

    toggleEkspanderYtelseIFokus() {
        const ytelseIFokus = this.props.ytelseIFokus;
        if (!ytelseIFokus) {
            return;
        }
        const erEkspandert = this.props.ekspanderteYtelser.includes(ytelseIFokus);
        this.props.ekspanderYtelse(ytelseIFokus, !erEkspandert);
    }

    handleSpace(event: React.KeyboardEvent) {
        if (eventTargetIsButton(event)) {
            return;
        }
        event.preventDefault();
        this.toggleEkspanderYtelseIFokus();
    }

    render() {
        return (
            <Wrapper onKeyDown={this.handleKeyDown} tabIndex={-1}>
                {this.props.children}
            </Wrapper>
        );
    }
}

function mapDispatchToProps(dispatch: Dispatch<AnyAction>): DispatchProps {
    return {
        settYtelseIFokus: ytelse => dispatch(setNyYtelseIFokus(ytelse)),
        ekspanderYtelse: (ytelse: Ytelse, ekspander: boolean) => dispatch(setEkspanderYtelse(ytelse, ekspander))
    };
}

function mapStateToProps(state: AppState): StateProps {
    return {
        ytelseIFokus: state.utbetalinger.ytelseIFokus,
        ekspanderteYtelser: state.utbetalinger.ekspanderteYtelser
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(HandleUtbetalingerHotKeys);
