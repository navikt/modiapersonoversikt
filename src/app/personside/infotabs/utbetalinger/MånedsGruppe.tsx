import * as React from 'react';
import styled from 'styled-components';
import theme from '../../../../styles/personOversiktTheme';
import { ArrayGroup } from '../../../../utils/groupArray';
import { Normaltekst } from 'nav-frontend-typografi';
import { Bold, Uppercase } from '../../../../components/common-styled-components';
import { getGjeldendeDatoForUtbetaling } from './utils/utbetalingerUtils';
import UtbetalingsKomponent from './utbetaling/Utbetaling';
import { Utbetaling } from '../../../../models/utbetalinger';

interface Props {
    gruppe: ArrayGroup<Utbetaling>;
}

const Wrapper = styled.li`
  > *:first-child {
    background-color: ${theme.color.kategori};
    padding: .2rem ${theme.margin.px20};
  }
  ol {
    padding: 0;
    margin: 0;
  }
  ol > *:not(:first-child) {
    border-top: ${theme.border.skille};
  }
`;

function Månedsgruppe({gruppe}: Props) {
    const utbetalingsKomponenter = gruppe.array.map(utbetaling => (
        <UtbetalingsKomponent
            key={getGjeldendeDatoForUtbetaling(utbetaling) + utbetaling.nettobeløp}
            utbetaling={utbetaling}
        />
    ));
    return (
        <Wrapper>
            <Normaltekst tag={'h3'}><Bold><Uppercase>{gruppe.category}</Uppercase></Bold></Normaltekst>
            <ol>
                {utbetalingsKomponenter}
            </ol>
        </Wrapper>
    );
}

export default Månedsgruppe;
