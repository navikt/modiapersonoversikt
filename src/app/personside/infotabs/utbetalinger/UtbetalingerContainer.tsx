import * as React from 'react';
import Utbetalinger from './Utbetalinger';
import { AppState } from '../../../../redux/reducers';
import { connect } from 'react-redux';
import { isLoading, isNotStarted, isReloading, Loaded, RestReducer } from '../../../../redux/restReducers/restReducer';
import { UtbetalingerResponse } from '../../../../models/utbetalinger';
import Innholdslaster from '../../../../components/Innholdslaster';
import { hentUtbetalinger, reloadUtbetalinger } from '../../../../redux/restReducers/utbetalinger';
import { default as Filtrering, FilterState, PeriodeValg } from './filter/Filter';
import { getFraDateFromFilter, getTilDateFromFilter } from './utils/utbetalingerUtils';
import theme from '../../../../styles/personOversiktTheme';
import styled from 'styled-components';
import ErrorBoundary from '../../../../components/ErrorBoundary';
import { loggEvent } from '../../../../utils/frontendLogger';
import Arenalenke from './Arenalenke/Arenalenke';
import { AsyncDispatch } from '../../../../redux/ThunkTypes';
import moment = require('moment');

interface State {
    filter: FilterState;
}

const initialState: State = {
    filter: {
        periode: {
            radioValg: PeriodeValg.SISTE_30_DAGER,
            egendefinertPeriode: {
                fra: moment().subtract(1, 'month').toDate(),
                til: new Date()
            }
        },
        utbetaltTil: [],
        ytelser: []
    }
};

interface StateProps {
    utbetalingerReducer: RestReducer<UtbetalingerResponse>;
}

interface DispatchProps {
    hentUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) => void;
    reloadUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) => void;
}

interface OwnProps {
    fødselsnummer: string;
}

type Props = StateProps & DispatchProps & OwnProps;

const UtbetalingerArticle = styled.article`
  display: flex;
  align-items: flex-start;
  @media(max-width: ${theme.media.utbetalinger}) {
    display: block;
  }
`;

const FiltreringSection = styled.section`
  min-width: 19rem;
  flex-basis: 19rem;
`;

const UtbetalingerSection = styled.section`
  position: relative;
  flex-grow: 1;
  min-width: 35rem; // Tabellene begynner å wrappe ved bredder mindre enn dette
  @media not all and (max-width: ${theme.media.utbetalinger}) {
      margin-left: ${theme.margin.layout};
  }
`;

class UtbetalingerContainer extends React.PureComponent<Props, State> {

    constructor(props: Props) {
        super(props);
        this.state = {...initialState};
        this.onFilterChange = this.onFilterChange.bind(this);
        this.reloadUtbetalinger = this.reloadUtbetalinger.bind(this);
        loggEvent('Sidevisning', 'Utbetalinger');
    }

    onFilterChange(change: Partial<FilterState>) {
        this.setState(
            (prevState) => { // Sender inn funksjon her for å motvirke race-conditions fra UtbetaltTilValg og YtelseValg
                return {
                    filter: {
                        ...prevState.filter,
                        ...change
                    }
                };
            }
        );
    }

    reloadUtbetalinger() {
        if (isLoading(this.props.utbetalingerReducer) || isReloading(this.props.utbetalingerReducer)) {
            return;
        }
        const fra = getFraDateFromFilter(this.state.filter);
        const til = getTilDateFromFilter(this.state.filter);
        this.props.reloadUtbetalinger(this.props.fødselsnummer, fra, til);
    }

    componentDidMount() {
        if (isNotStarted(this.props.utbetalingerReducer)) {
            this.props.hentUtbetalinger(
                this.props.fødselsnummer,
                getFraDateFromFilter(this.state.filter),
                getTilDateFromFilter(this.state.filter)
            );
        }
    }

    render() {
        return (
            <ErrorBoundary>
                <UtbetalingerArticle role="region" aria-label="Utbetalinger">
                    <div>
                        <Arenalenke fødselsnummer={this.props.fødselsnummer}/>
                        <FiltreringSection>
                            <Filtrering
                                filterState={this.state.filter}
                                onChange={this.onFilterChange}
                                hentUtbetalinger={this.reloadUtbetalinger}
                                utbetalingReducer={this.props.utbetalingerReducer}
                            />
                        </FiltreringSection>
                    </div>
                    <UtbetalingerSection aria-label="Filtrerte utbetalinger">
                        <Innholdslaster avhengigheter={[this.props.utbetalingerReducer]}>
                            <Utbetalinger
                                utbetalingerData={(this.props.utbetalingerReducer as Loaded<UtbetalingerResponse>).data}
                                filter={this.state.filter}
                            />
                        </Innholdslaster>
                    </UtbetalingerSection>
                </UtbetalingerArticle>
            </ErrorBoundary>
        );
    }
}

function mapStateToProps(state: AppState): StateProps {
    return ({
        utbetalingerReducer: state.restEndepunkter.utbetalingerReducer
    });
}

function mapDispatchToProps(dispatch: AsyncDispatch): DispatchProps {
    return {
        hentUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) =>
            dispatch(hentUtbetalinger(fødselsnummer, fra, til)),
        reloadUtbetalinger: (fødselsnummer: string, fra: Date, til: Date) =>
            dispatch(reloadUtbetalinger(fødselsnummer, fra, til))
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(UtbetalingerContainer);
