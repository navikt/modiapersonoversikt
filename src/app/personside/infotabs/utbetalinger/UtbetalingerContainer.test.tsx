import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../../test/Testprovider';
import UtbetalingerContainer from './UtbetalingerContainer';
import { getTestStore, mockReactQuery, setupReactQueryMocks } from '../../../../test/testStore';
import { statiskMockUtbetaling } from '../../../../mock/utbetalinger/statiskMockUtbetaling';
import FetchMock from 'yet-another-fetch-mock';
import MockDate from 'mockdate';
import utbetalingerResource from '../../../../rest/resources/utbetalingerResource';

test('Viser utbetalingercontainer med alt innhold', (done) => {
    MockDate.reset();
    setupReactQueryMocks();
    const mock = FetchMock.configure({
        enableFallback: false
    });
    mockReactQuery(utbetalingerResource.useFetch, {
        utbetalinger: [statiskMockUtbetaling],
        periode: { sluttDato: '1986-12-28', startDato: '1905-01-01' }
    });

    const testStore = getTestStore();

    const visittkortheader = renderer.create(
        <TestProvider customStore={testStore}>
            <UtbetalingerContainer />
        </TestProvider>
    );
    setTimeout(() => {
        const json = visittkortheader.toJSON();
        expect(json).toMatchSnapshot();
        mock.restore();
        done();
    }, 500);
});
