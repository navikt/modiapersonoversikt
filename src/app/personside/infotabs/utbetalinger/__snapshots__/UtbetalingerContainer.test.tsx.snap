// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Viser utbetalingercontainer med alt innhold 1`] = `
.c15 {
  border: none;
  background-color: transparent;
  padding: .1rem .2rem;
  border-radius: .35rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #0067c5;
}

.c15 svg {
  margin-left: .5rem;
  height: 1.2rem;
  width: 1.2rem;
}

.c15:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c15:hover {
  opacity: 0.8;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 .5rem rgba(0,0,0,.1);
  padding: 1.25rem;
  margin-bottom: .8rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 1.5rem;
}

.c5 > *:first-child {
  margin-bottom: .5rem;
}

.c5 > * {
  margin-top: .5rem;
}

.c7 {
  margin-top: 1rem;
}

.c6 {
  border: none;
  margin: 0;
  padding: 0;
}

.c6 > *:first-child {
  margin-bottom: .8rem;
}

.c6 > *:last-child {
  margin-bottom: 0;
}

.c30 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c27 {
  text-transform: uppercase;
}

.c26 {
  font-weight: bold;
}

.c18 {
  position: relative;
  border: none;
  padding: 0;
  border-radius: .35rem;
  cursor: pointer;
  background-color: transparent;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #0067c5;
}

.c18:after {
  border-bottom: 1px #B7B1A9 solid;
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100%;
}

.c18:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #FFBD66;
}

.c18:hover {
  opacity: 0.8;
}

.c18:hover:after {
  content: '';
  border-color: #0067C5;
}

.c20 {
  -webkit-transition: .5s;
  transition: .5s;
  padding: 0 1.25rem 1.25rem;
}

.c21 {
  -webkit-transition: .5s;
  transition: .5s;
  padding: 0 1.25rem;
}

.c19 {
  margin-right: .5rem;
}

.c16 {
  margin-right: 1.25rem;
}

.c11 table {
  width: 100%;
  text-align: right;
  border-spacing: 0;
}

.c11 table * {
  padding: 0;
  margin: 0;
}

.c11 table tr > * {
  padding: 0.1rem;
  text-align: right;
}

.c11 table tr > *:first-child {
  text-align: left;
}

.c11 table tr > *:not(:first-child) {
  width: 6rem;
}

.c10 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 .5rem rgba(0,0,0,.1);
  cursor: pointer;
}

.c12 {
  padding: 1.25rem 1.25rem 0;
}

.c13 {
  margin: 1rem 0;
}

.c13 th {
  font-weight: normal;
}

.c13 th {
  text-transform: uppercase;
}

.c13 td {
  font-weight: bold;
}

.c28 {
  cursor: pointer;
  -webkit-transition: .3s;
  transition: .3s;
}

.c28:focus {
  position: relative;
  outline: none;
}

.c28:focus::after {
  position: absolute;
  top: 0;
  content: '';
  box-shadow: 0 0 0 0.1875rem #FFBD66;
  height: 100%;
  width: 100%;
  pointer-events: none;
}

.c29 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0.625rem 1.25rem 0;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  cursor: pointer;
}

.c29 > *:nth-child(3) {
  margin-bottom: .8rem;
}

.c29 .order-first {
  -webkit-order: -1;
  -ms-flex-order: -1;
  order: -1;
}

.c25 > *:first-child {
  background-color: #cce1f3;
  padding: .2rem 1.25rem;
}

.c25 ol {
  padding: 0;
  margin: 0;
}

.c25 ol > *:not(:first-child) {
  border-top: solid 0.0625rem #b7b1a9;
}

.c22:focus {
  outline: none;
}

.c23 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 .5rem rgba(0,0,0,.1);
  margin-top: .8rem;
  margin-bottom: .8rem;
}

.c23 > *:first-child {
  padding: 1.25rem;
}

.c24 {
  padding: 0;
  margin: 0;
}

.c9 ol {
  list-style: none;
}

.c1 {
  border-radius: 0.25rem;
  background-color: white;
  box-shadow: 0 0 .5rem rgba(0,0,0,.1);
  font-weight: bold;
}

.c1 a {
  padding: 1.25rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  min-width: 19rem;
  -webkit-flex-basis: 19rem;
  -ms-flex-preferred-size: 19rem;
  flex-basis: 19rem;
}

.c8 {
  position: relative;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 35rem;
}

@media print {
  .c15 {
    display: none;
  }
}

@media (max-width:80rem) {
  .c4 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }

  .c4 > * {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    -webkit-flex-basis: 30%;
    -ms-flex-preferred-size: 30%;
    flex-basis: 30%;
  }

  .c4 > *:not(:last-child) {
    margin-right: 1rem;
  }
}

@media print {
  .c17 {
    display: none;
  }
}

@media print {
  .c28 {
    list-style-type: none;
  }
}

@media (max-width:80rem) {
  .c0 {
    display: block;
  }
}

@media not all and (max-width:80rem) {
  .c8 {
    margin-left: .8rem;
  }
}

<article
  aria-label="Utbetalinger"
  className="c0"
  role="region"
>
  <div>
    <nav
      className="c1"
    >
      <a
        className="lenkepanel"
        href="http://arena.adeo.no/forms/arenaMod.html?oppstart_skj=UB_22_MELDEHISTORIKK&fodselsnr=10108000398"
        rel="noopener noreferrer"
        target="_blank"
      >
        Meldekort i Arena
        <span
          className="lenkepanel__indikator"
        />
      </a>
    </nav>
    <section
      className="c2"
    >
      <nav
        aria-label="Filtrering utbetalinger"
        className="c3"
        onClick={[Function]}
      >
        <h2
          className="typo-undertittel"
        >
          Filtrering
        </h2>
        <div
          className="c4"
        >
          <form
            className="c5"
          >
            <fieldset
              className="c6"
            >
              <legend
                className="typo-etikett-liten"
              >
                Velg periode
              </legend>
              <div
                className="skjemaelement skjemaelement--horisontal"
              >
                <input
                  checked={true}
                  className="skjemaelement__input radioknapp"
                  id="Helt tilfeldig ID"
                  name="FiltreringsvalgGruppe"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Siste 30 dager
                </label>
              </div>
              <div
                className="skjemaelement skjemaelement--horisontal"
              >
                <input
                  checked={false}
                  className="skjemaelement__input radioknapp"
                  id="Helt tilfeldig ID"
                  name="FiltreringsvalgGruppe"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Inneværende år
                </label>
              </div>
              <div
                className="skjemaelement skjemaelement--horisontal"
              >
                <input
                  checked={false}
                  className="skjemaelement__input radioknapp"
                  id="Helt tilfeldig ID"
                  name="FiltreringsvalgGruppe"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  I fjor
                </label>
              </div>
              <div
                className="skjemaelement skjemaelement--horisontal"
              >
                <input
                  checked={false}
                  className="skjemaelement__input radioknapp"
                  id="Helt tilfeldig ID"
                  name="FiltreringsvalgGruppe"
                  onChange={[Function]}
                  type="radio"
                />
                <label
                  className="skjemaelement__label"
                  htmlFor="Helt tilfeldig ID"
                >
                  Egendefinert
                </label>
              </div>
            </fieldset>
            <div
              className="c7"
            >
              <button
                className="knapp knapp--standard"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Hent utbetalinger
              </button>
            </div>
          </form>
          <form
            className="c5"
          >
            <p
              className="typo-etikett-liten"
            >
              Utbetaling til
            </p>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={true}
                className="skjemaelement__input checkboks"
                id="Helt tilfeldig ID"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Bruker
              </label>
              <div
                aria-live="assertive"
                role="alert"
              />
            </div>
          </form>
          <form
            className="c5"
          >
            <p
              className="typo-etikett-liten"
            >
              Velg ytelse
            </p>
            <div
              className="skjemaelement skjemaelement--horisontal"
            >
              <input
                checked={true}
                className="skjemaelement__input checkboks"
                id="Helt tilfeldig ID"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Sykepenger
              </label>
              <div
                aria-live="assertive"
                role="alert"
              />
            </div>
          </form>
        </div>
      </nav>
    </section>
  </div>
  <section
    aria-label="Filtrerte utbetalinger"
    className="c8"
  >
    <div
      className="c9"
    >
      <article
        aria-label="Totalt utbetalt"
        className="c10"
        onClick={[Function]}
      >
        <div
          className="c11"
        >
          <div
            className="c12"
          >
            <h2
              className="typo-undertittel"
            >
              Totalt utbetalt for perioden
            </h2>
            <section
              className="c13"
            >
              <span
                className="typo-normal"
              >
                <table
                  role="table"
                >
                  <thead
                    role="rowgroup"
                  >
                    <tr
                      role="row"
                    >
                      <th
                        role="columnheader"
                      >
                        Totalt Utbetalt
                      </th>
                      <th
                        role="columnheader"
                      >
                        Brutto
                      </th>
                      <th
                        role="columnheader"
                      >
                        Trekk
                      </th>
                      <th
                        role="columnheader"
                      >
                        Utbetalt
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    role="rowgroup"
                  >
                    <tr
                      role="row"
                    >
                      <td
                        role="cell"
                      >
                        01.01.1905 - 28.12.1986
                      </td>
                      <td
                        role="cell"
                      >
                        13,370.00
                      </td>
                      <td
                        role="cell"
                      >
                        -1,984.00
                      </td>
                      <td
                        role="cell"
                      >
                        11,386.00
                      </td>
                    </tr>
                  </tbody>
                </table>
              </span>
            </section>
            <div
              className="c14"
            >
              <span>
                <button
                  aria-label="Skriv ut"
                  className="c15"
                  onClick={[Function]}
                >
                  <span
                    className="typo-normal"
                  >
                    Skriv ut
                  </span>
                   
                  <svg
                    viewBox="0 0 17 16"
                    xmlns="http://www.w3.org/2000/svg"
                    xmlnsXlink="http://www.w3.org/1999/xlink"
                  >
                    <path
                      d="M14.8,4.9H2.2c-0.9,0-1.7,0.7-1.7,1.7v4c0,0.9,0.7,1.7,1.7,1.7h1.7v3c0,0.2,0.1,0.3,0.3,0.3h8.7	c0.2,0,0.3-0.1,0.3-0.3v-3h1.7c0.9,0,1.7-0.7,1.7-1.7v-4C16.5,5.6,15.8,4.9,14.8,4.9z M4.5,14.9h8v-4.7h-8V14.9z"
                      fill="#0067C5"
                      fillRule="evenodd"
                    />
                    <path
                      d="M4.9,0.9h7.1c0.6,0,1,0.4,1,1v1.4c0,0.6-0.4,1-1,1H4.9c-0.6,0-1-0.4-1-1V1.9C3.9,1.3,4.4,0.9,4.9,0.9z"
                      fill="#0067C5"
                    />
                  </svg>
                </button>
              </span>
            </div>
          </div>
          <div
            className="c16"
          >
            <div
              className="c17"
            >
              <div
                className="c14"
              >
                <button
                  aria-expanded={false}
                  className="c18"
                  onClick={[Function]}
                >
                  <span
                    className="c19"
                  >
                    <span
                      className="typo-normal"
                    >
                      Vis
                       
                      sammendrag
                    </span>
                  </span>
                  <i
                    className="nav-frontend-chevron chevronboks chevron--ned"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            className="c20"
            open={false}
          >
            <div
              className="c21"
              open={false}
            />
          </div>
        </div>
      </article>
      <div
        className="c22"
        onKeyDown={[Function]}
        tabIndex={-1}
      >
        <article
          aria-label="Utbetalinger"
          className="c23"
        >
          <h2
            className="typo-undertittel"
          >
            Utbetalinger
          </h2>
          <ol
            aria-label="Måneder"
            className="c24"
          >
            <li
              className="c25"
            >
              <h3
                className="typo-normal"
              >
                <span
                  className="c26"
                >
                  <span
                    className="c27"
                  >
                    Mars 2017
                  </span>
                </span>
              </h3>
              <ol
                aria-label="Utbetalinger fra Mars 2017"
              >
                <li
                  className="c28"
                  onClick={[Function]}
                  onFocus={[Function]}
                  tabIndex={0}
                >
                  <article
                    aria-expanded={false}
                    aria-label="Utbetaling Sykepenger"
                  >
                    <div
                      className="c11"
                    >
                      <div
                        className="c29"
                      >
                        <div
                          className="c30"
                        >
                          <h4
                            className="typo-normal"
                          >
                            <span
                              className="c26"
                            >
                              Sykepenger
                            </span>
                          </h4>
                          <p
                            className="typo-normal"
                          >
                            <span
                              className="c26"
                            >
                              11,386.00
                            </span>
                          </p>
                        </div>
                        <p
                          className="typo-normal order-first"
                        >
                          29. Mars 2017
                           / 
                          <span
                            className="c26"
                          >
                            Utbetalt
                          </span>
                        </p>
                        <div
                          className="c30"
                        >
                          <p
                            className="typo-normal"
                          >
                            09.11.2016 - 31.10.2017
                          </p>
                          <p
                            className="typo-normal"
                          >
                            
                          </p>
                        </div>
                        <div
                          className="c30"
                        >
                          <p
                            className="typo-normal"
                          >
                            Utbetaling til: 
                            AREMARK TESTFAMILIEN
                          </p>
                          <span>
                            <button
                              aria-label="Skriv ut"
                              className="c15"
                              onClick={[Function]}
                            >
                              <span
                                className="typo-normal"
                              >
                                Skriv ut
                              </span>
                               
                              <svg
                                viewBox="0 0 17 16"
                                xmlns="http://www.w3.org/2000/svg"
                                xmlnsXlink="http://www.w3.org/1999/xlink"
                              >
                                <path
                                  d="M14.8,4.9H2.2c-0.9,0-1.7,0.7-1.7,1.7v4c0,0.9,0.7,1.7,1.7,1.7h1.7v3c0,0.2,0.1,0.3,0.3,0.3h8.7	c0.2,0,0.3-0.1,0.3-0.3v-3h1.7c0.9,0,1.7-0.7,1.7-1.7v-4C16.5,5.6,15.8,4.9,14.8,4.9z M4.5,14.9h8v-4.7h-8V14.9z"
                                  fill="#0067C5"
                                  fillRule="evenodd"
                                />
                                <path
                                  d="M4.9,0.9h7.1c0.6,0,1,0.4,1,1v1.4c0,0.6-0.4,1-1,1H4.9c-0.6,0-1-0.4-1-1V1.9C3.9,1.3,4.4,0.9,4.9,0.9z"
                                  fill="#0067C5"
                                />
                              </svg>
                            </button>
                          </span>
                        </div>
                      </div>
                      <div
                        className="c16"
                      >
                        <div
                          className="c17"
                        >
                          <div
                            className="c14"
                          >
                            <button
                              aria-expanded={false}
                              className="c18"
                              onClick={[Function]}
                            >
                              <span
                                className="c19"
                              >
                                <span
                                  className="typo-normal"
                                >
                                  Vis
                                   
                                  detaljer
                                </span>
                              </span>
                              <i
                                className="nav-frontend-chevron chevronboks chevron--ned"
                              />
                            </button>
                          </div>
                        </div>
                      </div>
                      <div
                        className="c20"
                        open={false}
                      >
                        <div
                          className="c21"
                          open={false}
                        />
                      </div>
                    </div>
                  </article>
                </li>
              </ol>
            </li>
          </ol>
        </article>
      </div>
    </div>
  </section>
</article>
`;
